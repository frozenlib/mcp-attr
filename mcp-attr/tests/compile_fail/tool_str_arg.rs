use mcp_attr::{
    server::{mcp_server, McpServer},
    Result,
};
struct InvalidArg;

struct MyServer;

#[mcp_server]
impl McpServer for MyServer {
    #[tool]
    async fn f(&self, a: &str) -> Result<String> {
        todo!()
    }
}

fn main() {}
