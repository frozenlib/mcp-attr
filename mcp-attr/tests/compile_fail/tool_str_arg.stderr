error: implementation of `serde::de::Deserialize` is not general enough
  --> tests/compile_fail/tool_str_arg.rs:12:23
   |
12 |     async fn f(&self, a: &str) -> Result<String> {
   |                       ^ implementation of `serde::de::Deserialize` is not general enough
   |
   = note: `&str` must implement `serde::de::Deserialize<'0>`, for any lifetime `'0`...
   = note: ...but `&str` actually implements `serde::de::Deserialize<'1>`, for some specific lifetime `'1`

warning: unused variable: `a`
  --> tests/compile_fail/tool_str_arg.rs:12:23
   |
12 |     async fn f(&self, a: &str) -> Result<String> {
   |                       ^ help: if this is intentional, prefix it with an underscore: `_a`
   |
   = note: `#[warn(unused_variables)]` on by default
